src/App.tsx

```diff
@@ -15,36 +16,108 @@
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = '',
+  ALPHABETICALLY = 'alphabetically',
+  LENGTH = 'length',
+}
+
+const getSortedGoods = (
+  goods: string[],
+  sortType: SortType,
+  reversed: boolean,
+): string[] => {
+  const sortedGoods = [...goods];
+
+  sortedGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABETICALLY:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortValue, setSortValue] = useState<SortType>(SortType.DEFAULT);
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  function sortGoods(type: SortType) {
+    setSortValue(type);
+  }
+
+  const toggleReverse = () => {
```

Comment:
The function `toggleReverse` is well-named as it starts with a verb, but for consistency with other event handlers, consider renaming it to something like `handleToggleReverse`.